src/storage/FirebaseStorageDirective.js

```diff
@@ -0,0 +1,32 @@
+/* istanbul ignore next */
+(function () {
+  "use strict";
+
+  function FirebaseStorageDirective($firebaseStorage, firebase) {
+    return {
+      restrict: 'A',
+      priority: 99, // run after the attributes are interpolated
+      scope: {},
+      link: function (scope, element, attrs) {
+        // $observe is like $watch but it waits for interpolation
+        // Ex: <img firebase-src="{{ myUrl }}"/>
+        attrs.$observe('firebaseSrc', function (newFirebaseSrcVal) {
+          if (newFirebaseSrcVal !== '' &&
```

Comment:
Actually, upon further testing and research I found out the `attrs.$observe` converts any value into a string. Therefore if you pass in an object, number, or bool it is automatically handled which makes the string check extra work. If you pass a `null` or `undefined` it is converted to an empty string, which is the only thing we need to check for.